pkg/installation/installation.go

```diff
@@ -0,0 +1,685 @@
+package installation
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/kyma-project/cli/cmd/kyma/version"
+	"github.com/kyma-project/cli/internal/helm"
+	"github.com/kyma-project/cli/internal/kube"
+	"github.com/kyma-project/cli/internal/kubectl"
+	"github.com/kyma-project/cli/internal/nice"
+	"github.com/kyma-project/cli/pkg/step"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/types"
+
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	releaseSrcURLPattern   = "https://raw.githubusercontent.com/kyma-project/kyma/%s/%s"
+	releaseResourcePattern = "https://raw.githubusercontent.com/kyma-project/kyma/%s/installation/resources/%s"
+	registryImagePattern   = "eu.gcr.io/kyma-project/kyma-installer:%s"
+	localDomain            = "kyma.local"
+)
+
+// Installation contains the installation elements and configuration options.
+type Installation struct {
+	k8s         kube.KymaKube
+	kubectl     *kubectl.Wrapper
```

Comment:
Introducing some interface instead of kubectl.Wrapper will make the design more flexible. The client could provide specific implementation.